### DISCUSSION TITLE
⏳ 🔗 Loop Grid > Add Responsive feature to "Items Per Page"

### DISCUSSION THREAD

**Discussion by erok415** [elementor/elementor#21923] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Items per page needs the responsive feature to determine how many items will show on a particular breakpoint.

### Describe the solution you'd like

Let's say you want to show 10 articles on desktop, but only want to show 1 or 2 articles on mobile. This is not possible with the current loop grid widget.

### Describe alternatives you've considered

The only alternative would be to hide articles with CSS.

**Comment by alfred1000351** [comment]
Have you found a nice way of handling this? feels like this should be a key thing to have

**Reply by erok415** [reply]
I believe when I reported it, I used the default rather than trying to find a fix as it was a low priority.

**Comment by saawood13** [comment]
I am also looking for a way to handle this

**Comment by ayoubkhan558** [comment]
I think it does not make sense.

**Reply by phkoon** [reply]
An applicable use case is what I'm facing right now: if you want a grid with 4 columns on desktop displaying 12 items, but in mobile it requires displaying only 1 or 2 columns, then it will display a humongous 12 rows section; resulting in horrible UI and even more horrible UX

**Reply by erok415** [reply]
That's correct @phkoon

**Comment by phkoon** [comment]
Much needed. Current "solution" is: for every breakpoint you must create a copy of the grid and hide the others, B L O A T

**Comment by Eetheart** [comment]
Please, can we have this?

**Comment by dsz36** [comment]
An other solution for this is to use a different loop grid element for all breakpoint, and you hide or show them with responsive advanced feature of elementor pro.

**Comment by xgrid-ayoubkhan558** [comment]
I don't think it makes any sense, For example if there is responsive Items per page, how you you will handle different pagination for different devices?

**Comment by DanzasDigital** [comment]
+1 I also need this exact feature

**Comment by spasauna** [comment]
+1

**Comment by memdigital** [comment]
+1 - facing this right now. Would be great to set items per page per breakpoint for better control over aesthetics.

**Reply by memdigital** [reply]
For those interested, I had Claude AI come up with a workaround, which I have tweaked to my own requirements and it works just fine. (Although, mine is a lot more condensed for my use case)... See instructions at bottom on how to use this.

/**
 * ========================================
 * Generic Responsive Posts Per Page for Elementor Loop Grids
 * 
 * Usage: Add CSS class to your Loop Grid widget in Elementor:
 * - 'responsive-posts-12' for 12/9/6/3 posts per breakpoint
 * - 'responsive-posts-9' for 9/6/4/2 posts per breakpoint
 * - 'responsive-posts-6' for 6/4/3/2 posts per breakpoint
 * - 'responsive-posts-4' for 4/3/2/1 posts per breakpoint
 * 
 * Standard Elementor Breakpoints:
 * Desktop: 1025px+, Tablet: 768-1024px, Mobile: 0-767px
 * ========================================
 */

/* 12/9/6/3 Pattern - Good for large grids */
@media (min-width: 1025px) {
    .responsive-posts-12 .e-loop-item:nth-child(n+13) { display: none !important; }
}
@media (min-width: 768px) and (max-width: 1024px) {
    .responsive-posts-12 .e-loop-item:nth-child(n+10) { display: none !important; }
}
@media (max-width: 767px) {
    .responsive-posts-12 .e-loop-item:nth-child(n+4) { display: none !important; }
}

/* 9/6/4/2 Pattern - Good for medium grids */
@media (min-width: 1025px) {
    .responsive-posts-9 .e-loop-item:nth-child(n+10) { display: none !important; }
}
@media (min-width: 768px) and (max-width: 1024px) {
    .responsive-posts-9 .e-loop-item:nth-child(n+7) { display: none !important; }
}
@media (min-width: 481px) and (max-width: 767px) {
    .responsive-posts-9 .e-loop-item:nth-child(n+5) { display: none !important; }
}
@media (max-width: 480px) {
    .responsive-posts-9 .e-loop-item:nth-child(n+3) { display: none !important; }
}

/* 6/4/3/2 Pattern - Good for smaller grids */
@media (min-width: 1025px) {
    .responsive-posts-6 .e-loop-item:nth-child(n+7) { display: none !important; }
}
@media (min-width: 768px) and (max-width: 1024px) {
    .responsive-posts-6 .e-loop-item:nth-child(n+5) { display: none !important; }
}
@media (min-width: 481px) and (max-width: 767px) {
    .responsive-posts-6 .e-loop-item:nth-child(n+4) { display: none !important; }
}
@media (max-width: 480px) {
    .responsive-posts-6 .e-loop-item:nth-child(n+3) { display: none !important; }
}

/* 4/3/2/1 Pattern - Good for minimal grids */
@media (min-width: 1025px) {
    .responsive-posts-4 .e-loop-item:nth-child(n+5) { display: none !important; }
}
@media (min-width: 768px) and (max-width: 1024px) {
    .responsive-posts-4 .e-loop-item:nth-child(n+4) { display: none !important; }
}
@media (min-width: 481px) and (max-width: 767px) {
    .responsive-posts-4 .e-loop-item:nth-child(n+3) { display: none !important; }
}
@media (max-width: 480px) {
    .responsive-posts-4 .e-loop-item:nth-child(n+2) { display: none !important; }
}

/* Optional: Smooth transitions */
[class*="responsive-posts-"] .e-loop-item {
    transition: opacity 0.3s ease;
}

/* Optional: Hide pagination on mobile for cleaner look */
@media (max-width: 767px) {
    [class*="responsive-posts-"] .elementor-pagination {
        /* display: none; */
    }
}

/**
 * ========================================
 * SETUP INSTRUCTIONS:
 * 1. Set your Loop Grid to show the MAXIMUM number of posts you want (12, 9, 6, or 4)
 * 2. Add the corresponding CSS class to your widget:
 *    - For 12 max posts: add 'responsive-posts-12'
 *    - For 9 max posts: add 'responsive-posts-9'
 *    - For 6 max posts: add 'responsive-posts-6'
 *    - For 4 max posts: add 'responsive-posts-4'
 * 3. The CSS will automatically hide excess posts on smaller screens
 * 
 * CUSTOMIZATION:
 * To create your own pattern, copy any section above and modify:
 * - Change the class name (e.g., 'responsive-posts-custom')
 * - Adjust the :nth-child(n+X) numbers to show different amounts
 * - Modify breakpoints if needed
 * ========================================
 */

**Reply by rutger-dezons** [reply]
With custom css you can do a lot here, as @memdigital answered with AI. https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/:nth-child 
```
// I'm using this to display 4 items on mobile, and use the defaults on tablet and desktop. This should work with any Elementor loop grid. 
selector { 
  .elementor-loop-container {
    .e-loop-item:nth-child(n+5) { 
      @media screen and (max-width: 768px){
        display: none !important; 
      }
    }
  }
}
```
Perhaps the dev team could use something like this for a quick and easy solution, though it would be better for them to solve this properly without a bunch of hidden content on the DOM.

**Reply by phkoon** [reply]
> With custom css you can do a lot here, as @memdigital answered with AI. https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Selectors/:nth-child
> 
> ```
> // I'm using this to display 4 items on mobile, and use the defaults on tablet and desktop. This should work with any Elementor loop grid. 
> selector { 
>   .elementor-loop-container {
>     .e-loop-item:nth-child(n+5) { 
>       @media screen and (max-width: 768px){
>         display: none !important; 
>       }
>     }
>   }
> }
> ```
> 
> Perhaps the dev team could use something like this for a quick and easy solution, though it would be better for them to solve this properly without a bunch of hidden content on the DOM.

Actually, it should be `max-width: 767px` for mobile. Aside from that, that's a good workaround
